GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -156,60 +165,55 @@ private void UICommands_PostRepositoryChanged(object sender, GitUIPluginInterfac
             public bool IgnoreSelectionChangedEvent { get; set; }
             protected GitModule Module => UICommands.Module;
 
+            public void CancelReloadNodes()
+            {
+                _reloadCancellationTokenSequence.CancelCurrent();
+            }
+
             // Invoke from child class to reload nodes for the current Tree. Clears Nodes, invokes
             // input async function that should populate Nodes, then fills the tree view with its contents,
             // making sure to disable/enable the control.
-            protected void ReloadNodes(Func<CancellationToken, Task> loadNodesTask)
+            protected async Task ReloadNodesAsync(Func<CancellationToken, Task> loadNodesTask)
             {
-                ThreadHelper.ThrowIfNotOnUIThread();
-
-                ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
-                {
-                    var repoObjectTree = TreeViewNode.TreeView.Parent;
-
-                    try
-                    {
-                        var token = _reloadCancellationTokenSequence.Next();
+                await TaskScheduler.Default;
 
-                        repoObjectTree.Enabled = false;
+                Nodes.Clear();
 
-                        TreeViewNode.TreeView.BeginUpdate();
-                        IgnoreSelectionChangedEvent = true;
-                        Nodes.Clear();
+                var token = _reloadCancellationTokenSequence.Next();
```

Comment:
`_reloadCancellationTokenSequence` needs to be disposed somewhere.

Reply:
Yes, forgot about that one.